leapsectz/leapsectz.go

```diff
@@ -88,34 +143,181 @@ func parse(r io.Reader) ([]LeapSecond, error) {
 		var nn uint32
 		err := binary.Read(r, binary.BigEndian, &nn)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		n[i] = int(nn)
 	}
 
 	// skip uninteresting data:
-	//  tzh_timecnt (char [4])s  coded transition times a la time(2)
+	//  tzh_timecnt (char [4] or char [8] for ver 2)s  coded transition times a la time(2)
 	//  tzh_timecnt (unsigned char)s types of local time starting at above
 	//  tzh_typecnt repetitions of
 	//   one (char [4])  coded UT offset in seconds
 	//   one (unsigned char) used to set tm_isdst
 	//   one (unsigned char) that's an abbreviation list index
 	//  tzh_charcnt (char)s  '\0'-terminated zone abbreviations
-	skip := n[NTime]*5 + n[NZone]*6 + n[NChar]
-	data := make([]byte, skip)
-	if n, _ := r.Read(data); n != skip {
-		return nil, errBadData
+	var skip int
+	if version == 0 {
+		skip = n[NTime]*5 + n[NZone]*6 + n[NChar]
+	} else {
+		skip = n[NTime]*9 + n[NZone]*6 + n[NChar]
+	}
+	if n, _ := io.CopyN(ioutil.Discard, r, int64(skip)); n != int64(skip) {
```

Comment:
Alternatively you can say that `r` is `ReadSeekCloser`, and use `Seek` function.